From 0e1be40a45d767a07c1fb50ebfba273368f57484 Mon Sep 17 00:00:00 2001
From: Beni Lev <beni.lev@intel.com>
Date: Thu, 28 Sep 2017 12:10:55 +0300
Subject: [PATCH] iwlwifi: mvm: allow reading UMAC error data from SMEM in A000 devices
Git-commit: 0e1be40a45d767a07c1fb50ebfba273368f57484
Patch-mainline: v4.15-rc1
References: FATE#326294

Currently, UMAC error data reading is restricted to DCCM.
A000 NICs use SMEM for this data.

Signed-off-by: Beni Lev <beni.lev@intel.com>
Signed-off-by: Luca Coelho <luciano.coelho@intel.com>
Acked-by: Takashi Iwai <tiwai@suse.de>

---
 drivers/net/wireless/intel/iwlwifi/mvm/utils.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/net/wireless/intel/iwlwifi/mvm/utils.c b/drivers/net/wireless/intel/iwlwifi/mvm/utils.c
index 2da1b088ac01..4ade688db63b 100644
--- a/drivers/net/wireless/intel/iwlwifi/mvm/utils.c
+++ b/drivers/net/wireless/intel/iwlwifi/mvm/utils.c
@@ -459,7 +459,8 @@ static void iwl_mvm_dump_umac_error_log(struct iwl_mvm *mvm)
 
 	base = mvm->umac_error_event_table;
 
-	if (base < 0x800000) {
+	if (base < (mvm->trans->cfg->device_family == IWL_DEVICE_FAMILY_A000 ?
+		    0x400000 : 0x800000)) {
 		IWL_ERR(mvm,
 			"Not valid error log pointer 0x%08X for %s uCode\n",
 			base,
-- 
2.19.2

